### ISSUE TITLE
Resolve binary submission issues for 6.1

### ISSUE THREAD

**Issue by astralbodies** [wordpress-mobile/WordPress-iOS#5081]
Three errors are preventing a binary from being submitted to Apple:

> ERROR ITMS-90535: "Unexpected CFBundleExecutable Key. The bundle at 'Payload/WordPress.app/HsUIResourceBundle.bundle' does not contain a bundle executable. If this bundle intentionally does not contain an executable, consider removing the CFBundleExecutable key from its Info.plist and using a CFBundlePackageType of BNDL. If this bundle is part of a third-party framework, consider contacting the developer of the framework for an update to address this issue."

This can be resolved by manually fixing HelpShift 5.5.0's resource bundle and removing the key. [See details on this "fix"](https://developers.helpshift.com/ios/getting-started/#swift).

> ERROR ITMS-90205: "Invalid Bundle. The bundle at 'WordPress.app/PlugIns/WordPressTodayWidget.appex' contains disallowed nested bundles."

> ERROR ITMS-90206: "Invalid Bundle. The bundle at 'WordPress.app/PlugIns/WordPressTodayWidget.appex' contains disallowed file 'Frameworks'."

The last two are related to CocoaPods. Investigate a fix and apply. The stats framework has a single Swift file that is complicating the submission process. Rather than switch back to Objective-C only, find a permanent solution to allow for a mixed source framework.

cc: @sendhil

**Comment by astralbodies** [comment]
Possible solution for the nested bundles issue: https://github.com/CocoaPods/CocoaPods/issues/4203#issuecomment-167162771

**Comment by sendhil** [comment]
@astralbodies does that fix work for our project?

**Comment by benasher44** [comment]
Just saw you linked the CocoaPods issue. I'd expect that adding a build phase to remove the frameworks folder from the extension after the CocoaPods embed build phase to work in combination with ensuring that your Runpath Search Paths for your extension target are setup correctly, such that it can find the frameworks it needs.

**Comment by astralbodies** [comment]
Thanks for the tip @benasher44! Luckily the framework at odds with the Today Extension is also included in the main app. I think we're going to have to add some search path changes in for the next release as we have a share extension target there. The share extension has at least one framework not part of the main app.

**Comment by benasher44** [comment]
@astralbodies welp, then your build phase will likely need to be a bit more complex that just `rm -rf Frameworks` :(. If it were me, I'd `ls` the frameworks in each directory and use the `comm` command to figure out which ones need to be copied up into the main frameworks embed location with the rest of them. I'll add this case on to the issue.

**Comment by astralbodies** [comment]
@benasher44 In the PR for this issue I noted that turning off `EMBEDDED_CONTENT_CONTAINS_SWIFT` in the today extension is now of course creating CocoaPods warnings. Can you think of a better way to remove the Swift libs Xcode includes in the extension bundle after the build instead of setting this build option to false?

**Comment by benasher44** [comment]
@astralbodies at the moment no. We just add a build phase that removes the frameworks folder after CocoaPods creates it via the embed script. What are the warnings you're getting?

**Comment by astralbodies** [comment]
@benasher44 Oops forgot to include them:
```
[!] The `WordPressTodayWidget [Debug]` target overrides the `EMBEDDED_CONTENT_CONTAINS_SWIFT` build setting defined in `../Pods/Target Support Files/Pods-WordPressTodayWidget/Pods-WordPressTodayWidget.debug.xcconfig'. This can lead to problems with the CocoaPods installation
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.

[!] The `WordPressTodayWidget [Release]` target overrides the `EMBEDDED_CONTENT_CONTAINS_SWIFT` build setting defined in `../Pods/Target Support Files/Pods-WordPressTodayWidget/Pods-WordPressTodayWidget.release.xcconfig'. This can lead to problems with the CocoaPods installation
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.

[!] The `WordPressTodayWidget [Release-Internal]` target overrides the `EMBEDDED_CONTENT_CONTAINS_SWIFT` build setting defined in `../Pods/Target Support Files/Pods-WordPressTodayWidget/Pods-WordPressTodayWidget.release-internal.xcconfig'. This can lead to problems with the CocoaPods installation
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.

[!] The `WordPressTodayWidget [Release-Alpha]` target overrides the `EMBEDDED_CONTENT_CONTAINS_SWIFT` build setting defined in `../Pods/Target Support Files/Pods-WordPressTodayWidget/Pods-WordPressTodayWidget.release-alpha.xcconfig'. This can lead to problems with the CocoaPods installation
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.
```

**Comment by benasher44** [comment]
Hm. I'm kind of confused by this one. I'd still recommend adding a build phase to explicitly fix Frameworks folder issues that runs after CocoaPod's scripts. I'm not sure why changing this flag tricks CocoaPods into not embedding frameworks into the extension target.

**Comment by astralbodies** [comment]
@benasher44 If I leave this flag set to true (coming from the CocoaPods config) then the Swift libraries get pushed into the Frameworks folder AFTER the build phases are done. So my build phase change deletes the Frameworks folder and then Xcode creates it again with a copy of all of the Swift runtimes. This ends up failing Apple's binary checks still.

**Comment by benasher44** [comment]
@astralbodies that's pretty odd. I've never seen this before. I wouldn't expect Xcode to add things to your project that would cause the build to be rejected. For the CocoaPods warnings, you could add `EMBEDDED_CONTENT_CONTAINS_SWIFT` in a post install hook, and I don't think CocoaPods would be upset about that.

**Comment by benasher44** [comment]
@astralbodies I had another thought about this one. Maybe if you have any 3rd-party libs in your "Link Binary with Libaries" phase for your extension, then I think this could happen? I would double check there.

**Comment by astralbodies** [comment]
Closing this for now as it's been resolved for 6.1 - may open a longer term issue regarding the discussions we've had above.
